dask/array/core.py

```diff
@@ -2077,6 +2093,10 @@ def __abs__(self):
 
     @check_if_handled_given_other
     def __add__(self, other):
+        from ..dataframe import DataFrame, Series
+
+        if isinstance(other, (DataFrame, Series)):
+            return NotImplemented
```

Comment:
Can this be moved to a function? Seems it's repeated quite a bit.

Reply:
Yea, good idea.

This is definitely one of the parts I wanted to come back to, make sure I understand all the semantics of better, and document. There are apparently some subtleties around which things `return NotImplemented` in order to delegate to which other things.

Reply:
alright, I factored this out to a `_elemwise_binary_df_op` helper. A couple other ops look similar but don't use `elemwise`, so I've left them as-is (`__matmul__`, `__divmod__`). Didn't see a way to include them in the DRYing w/o over-complicating things.

I'm not 100% sure about which ops should do this "delegate to _Frame`" vs. not; I added it to left-associative ops (tests were surfacing problems as I implemented previously-unimplemented code-paths related to `iloc`) but not their right-associative counterparts. I don't know if that's correct/complete; need to dig in to the details more.